src/scanpy/preprocessing/_simple.py

```diff
@@ -637,6 +637,22 @@ def normalize_per_cell(
 DT = TypeVar("DT")
 
 
+@njit
+def _create_regressor_categorical(
+    X: np.ndarray, number_categories: int, cat_array: np.ndarray
+) -> np.ndarray:
+    # create regressor matrix for categorical variables
+    regressors = np.zeros(X.shape, dtype=X.dtype)
```

Comment:
check dtype for behavior with integer dtype i.e., need to ensure this is a floating point matrix